components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/dao/AppCustomContentDAO.java

```diff
@@ -0,0 +1,253 @@
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTableColumns.APP_ID;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTableColumns.CONTENT;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTableColumns.CONTENT_TYPE;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTableColumns.TENANT_ID;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTypes.CONTENT_TYPE_CSS;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTypes.CONTENT_TYPE_HTML;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.DaoConstants.CustomContentTypes.CONTENT_TYPE_JS;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.SQLConstants.DELETE_APP_CUSTOM_CONTENT_SQL;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.SQLConstants.GET_APP_CUSTOM_CONTENT_COUNT_SQL;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.SQLConstants.GET_APP_CUSTOM_CONTENT_SQL;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.SQLConstants.INSERT_APP_CUSTOM_CONTENT_SQL;
+import static org.wso2.carbon.identity.branding.preference.management.core.dao.constants.SQLConstants.UPDATE_APP_CUSTOM_CONTENT_SQL;
+
+/**
+ * This class is to perform CRUD operations for Application vise Custom Content
+ */
+
+public class AppCustomContentDAO {
+
+    /**
+     * Checks whether custom content exists for the given APP.
+     *
+     * @param applicationUuid   Application UUID.
+     * @param tenantId          Tenant ID.
+     * @return                  {@code true} if custom content exists for the tenant, {@code false} otherwise.
+     * @throws CustomContentServerException if an error occurs during the database access.
+     */
+    public boolean isAppCustomContentAvailable(String applicationUuid, int tenantId)
+            throws CustomContentServerException {
+
+        NamedJdbcTemplate template = JdbcUtils.getNewNamedJdbcTemplate();
+        try {
+            Integer count = template.fetchSingleRecord(GET_APP_CUSTOM_CONTENT_COUNT_SQL,
+                    (resultSet, rowNum) -> resultSet.getInt(1),
+                    namedPreparedStatement -> {
+                        namedPreparedStatement.setString(1, applicationUuid);
+                        namedPreparedStatement.setInt(2, tenantId);
+                    });
+            return count != null && count > 0;
+        } catch (DataAccessException e) {
+            throw new CustomContentServerException("Error checking if custom content exists for application " +
```

Comment:
Shall we define error constants for all the possible error scenarios with proper error codes and messages? You may refer this https://github.com/wso2-extensions/identity-branding-preference-management/blob/main/components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/constant/BrandingPreferenceMgtConstants.java#L53